src/controllers/attachment/index.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  Security,
+  Tags,
+  UploadedFile,
+  Request,
+  Controller,
+  Get,
+  Route,
+  Path,
+  Post,
+  Header,
+  Response,
+  SuccessResponse,
+} from 'tsoa'
+import { Logger } from 'pino'
+import express from 'express'
+
+import { logger } from '../../lib/logger'
+import Database, { Models, Query } from '../../lib/db'
+import type { Attachment } from '../../models'
+import { BadRequest, NotFound } from '../../lib/error-handler'
+
+type File = {
+  [k: string]: string
+}
+
+@Route('attachment')
+@Tags('attachment')
+@Security('bearerAuth')
+export class attachment extends Controller {
+  log: Logger
+  dbClient: Database = new Database()
+  db: Models<Query> | any // TMP this is the only one could not address now
+
+  constructor() {
+    super()
+    this.log = logger.child({ controller: '/attachment' })
+    // TMP will be updated with a wrapper so ORM client independant
+    this.db = this.dbClient.db()
+  }
+
+  @Get('/')
+  @SuccessResponse(200, 'returns all attachments')
+  public async get(): Promise<Attachment[]> {
+    this.log.debug('retrieving all attachments')
+    const result = await this.db.attachment()
+
+    return result.map(({ binary_blob, ...item }: Attachment) => ({
+      ...item,
+      size: binary_blob.size,
+    }))
+  }
+
+  @Post('/')
+  @SuccessResponse(201, 'attachment has been created')
+  public async create(@Request() req: express.Request, @UploadedFile() file: File): Promise<{ [k: string]: string }> {
```

Comment:
this should be able to just return an `Attachment` right?

Reply:
Mmm.. yes according to knex [insert](https://knexjs.org/guide/query-builder.html#insert) and [returning](https://knexjs.org/guide/query-builder.html#returning) docs added returning most of the props, probly could be taken from payload, but i think from db makes more sense.

This is another reason why we need to have generic wrappers for ORM e.g. getBy etc